File: Atcoder_#306/javascript/c_centers.js
function main(input) {
    const lines = input.trim().split('\n');

    // Nの取得
    const n = parseInt(lines[0]);

    // 数列Aの取得
    const a = lines[1].split(' ').map(Number);

    // 各数字の出現位置を記録するオブジェクト
    const positionMap = {};

    // 各要素の出現位置を記録
    a.forEach((num, idx) => {
        if (!positionMap[num]) {
            positionMap[num] = [];
        }
        positionMap[num].push(idx + 1);  // 1-based index
    });

    // f(i)のリストを作成し、2番目の出現位置でソート
    const result = Array.from({ length: n }, (_, i) => i + 1)
        .sort((i, j) => positionMap[i][1] - positionMap[j][1]);

    // 結果を出力
    console.log(result.join(' '));
}

// 標準入力を読み込み、main関数を呼び出し
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin', 'utf8');
main(input);

File: Atcoder_#306/ruby/c_centers.rb
def main
  # 入力を取得
  n = gets.to_i
  a = gets.split.map(&:to_i)

  # 各数字の2番目の出現位置を記録するハッシュ
  position_map = Hash.new { |hash, key| hash[key] = [] }

  # 各要素の出現位置を記録
  a.each.with_index do |num, idx|
    position_map[num] << idx + 1
  end

  # f(i) のリストを作成して、2番目の出現位置でソート
  result = (1..n).to_a.sort_by { |i| position_map[i][1] }

  # 結果を出力
  puts result.join(' ')
end

main
